token-erc-721/chaincode-java/src/main/java/org/hyperledger/fabric/samples/erc721/models/NFT.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.fabric.samples.erc721.models;
+
+import java.io.Serializable;
+
+import org.hyperledger.fabric.contract.annotation.DataType;
+import org.hyperledger.fabric.contract.annotation.Property;
+import org.json.JSONObject;
+
+import com.owlike.genson.Genson;
+import com.owlike.genson.annotation.JsonProperty;
+
+@DataType()
+public final class NFT implements Serializable {

```

Comment:
where can I find Nun-fungibilities on this model?

Reply:
In this sample, it is assumed that only one organization (played by Org1) is in an issuer role and can mint new tokens into their account. The mintWithTokenURI smart contract function reads the certificate information of the client identity that submitted the transaction using the GetClientIdentity.GetID() API and creates a non-fungible token associated with the client ID with the requested token ID.  The token ID  is the unique identifier that ensures non-fungibility.